internal/cli/tiltfile_result.go

```diff
@@ -0,0 +1,75 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/windmilleng/tilt/internal/analytics"
+	"github.com/windmilleng/tilt/internal/tiltfile"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+// arbitrary non-1 value chosen to allow callers to distinguish between
+// Tilt errors and Tiltfile errors
+const TiltfileErrExitCode = 5
+
+type tiltfileResultCmd struct {
+	fileName string
+}
+
+var _ tiltCmd = &tiltfileResultCmd{}
+
+type cmdTiltfileResultDeps struct {
+	tfl tiltfile.TiltfileLoader
+}
+
+func newTiltfileResultDeps(tfl tiltfile.TiltfileLoader) cmdTiltfileResultDeps {
+	return cmdTiltfileResultDeps{
+		tfl: tfl,
+	}
+}
+
+func newTiltfileResultCmd() *tiltfileResultCmd {
+	return &tiltfileResultCmd{}
+}
+
+func (c *tiltfileResultCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tiltfile-result",
+		Short: "Exec the Tiltfile and print results as JSON (note: the API is unstable and may change)",
+	}
+
+	cmd.Flags().StringVar(&c.fileName, "file", tiltfile.FileName, "Path to Tiltfile")
+
+	return cmd
+}
+
+func (c *tiltfileResultCmd) run(ctx context.Context, args []string) error {
+	deps, err := wireTiltfileResult(ctx, analytics.Get(ctx))
+	if err != nil {
+		return wrapUnexpectedError(errors.Wrap(err, "wiring dependencies"))
+	}
+
+	tlr := deps.tfl.Load(ctx, c.fileName, model.NewUserConfigState(args))
+	if tlr.Error != nil {
+		// Some errors won't JSONify properly by default, so just print it
+		// to STDERR and use the exit code to indicate that it's an error
+		// from Tiltfile parsing.
+		fmt.Fprintln(os.Stderr, tlr.Error)
+		os.Exit(TiltfileErrExitCode)
+	}
+
+	err = encodeJSON(tlr, os.Stderr)
```

Comment:
can you elaborate why we'd want this output to go to stderr? 

I expect most people would expect it to go to stdout, like it does for other unix command-line tools (and for our commands). That way you can pipe it to other tools, e.g.

```
tilt dump engine | jq ".TiltStartTime"
```

Reply:
+1

Reply:
similarly, to make this flow cleaner, I wonder if it'd make sense to, change tiltfile output:
1. don't print it by default
2. print it to stderr on failure
3. print it to stderr if some flag is set (though I couldn't come up with a name in 15 seconds, so who knows)

Reply:
Ya, my first instinct is that any Tiltfile logs or warnings would go to stderr

Reply:
I can't remember from where but I had an idea of putting structured output into stderr? But yeah Matt's plan sgtm